### DISCUSSION TITLE
I'd like to connect to a message broker or queue system

### DISCUSSION THREAD

**Discussion by Salles-FA** [nextgenhealthcare/connect#5091] [General]
Hi,

I'm using Mirth Connect as an option to connect a multi-channel health care device with some message broker or queue system. This is an academic project for a post graduation I'm in.

I'm searching and struggling to find some documentation regarding this. I only found an old article explaining how to connect to an Azure Queue Store, but it isn't working with the current version of Mirth Connect.

- https://www.spheregen.com/mirth-connect-with-microsoft-azure-queue-storage/

Could you please give me any tips?

It could be to AWS SQS, Kafka, RabbitMQ... Or any other one.

I thank you in advance,
Fábio

**Comment by pacmano1** [comment]
If you are new to Mirth, I would use a broker that supports REST calls, that's standard functionality in Mirth.

**Reply by Salles-FA** [reply]
Do you have an example? Wiki? Tutorial? Reference documentation?

Thank you!

**Reply by pacmano1** [reply]
Calling APIs with http senders is a fundamental capability in Mirth so I don't know what you are asking.  Are you new to mirth?   Please see https://github.com/nextgenhealthcare/connect/discussions/5030 if you are new to mirth, there are recommendations on learning materials.

**Reply by Salles-FA** [reply]
Hi @pacmano1 ,

Yes, I'm totally new to Mirth. I'd like some good material showing how to connect it to a message broker... Mainly a cloud native one.

Could you please lend me a hand?

**Reply by pacmano1** [reply]
IMHO, there is much ground work you need to know to get started with Mirth and not attending some basic training or reading the material proposed on the other discussion thread make that hard.   So no, I can't spend the time to do that but others hopefully will jump in.    Another frequent poster mentioned exploring the JMS sender and receiver.   

Also, while this is not the mirth forums, the etiquette guide over there is useful.   Please see https://forums.mirthproject.io/articles/177082-forum-etiquette.  In particular, bullets 8, 9 and 10.

**Comment by thorst** [comment]
Right, learn mirth first, and if you dont know about apis, learn about them, then use the client apis in mirth to get done what you need done. We also have some custom apis using a channel that returns json, to supplement the client apis as needed, so you would do the same to get anything done you need to.
